From a04923b681da5f734d8462cfcc129db30bf9b3c1 Mon Sep 17 00:00:00 2001
From: rkcosta <rkcosta@283d02a7-25f6-0310-bc7c-ecb5cbfe19da>
Date: Mon, 11 Jul 2011 03:23:39 +0000
Subject: [PATCH] SVN commit 1240907 by rkcosta:
Upstream: Yes. In 4.6, 4.7 and trunk branches.

cliinterface: Do not assume a process does not exist in runProcess().

The Q_ASSERT(!m_process) in the beginning of runProcess() was wrong, as when
one is adding files to an archive (or creating an archive) processFinished()
calls list(), which in turn calls runProcess() as well, thus making the assert
fail.

The best thing to do is just wait for the existing process to finish before
launching the new one.

Backport of r1240907, in case the 4.6 ever gets packaged again.

CCBUG: 277393


git-svn-id: svn://anonsvn.kde.org/home/kde/branches/KDE/4.6/kdeutils@1240908 283d02a7-25f6-0310-bc7c-ecb5cbfe19da
---
 ark/kerfuffle/cliinterface.cpp |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/ark/kerfuffle/cliinterface.cpp b/ark/kerfuffle/cliinterface.cpp
index 6ff72c8..8a059c8 100644
--- a/ark/kerfuffle/cliinterface.cpp
+++ b/ark/kerfuffle/cliinterface.cpp
@@ -325,10 +325,13 @@ bool CliInterface::runProcess(const QString& programName, const QStringList& arg
         return false;
     }
 
-    Q_ASSERT(!m_process);
-
     kDebug() << "Executing" << programPath << arguments;
 
+    if (m_process) {
+        m_process->waitForFinished();
+        delete m_process;
+    }
+
 #ifdef Q_OS_WIN
     m_process = new KProcess();
 #else
-- 
1.7.6

